test/clients/fermionFractions.ts

```diff
@@ -926,6 +927,153 @@ describe("FermionFNFT - fractionalisation tests", function () {
+    };
+    const custodianFee = {
+      amount: parseEther("0.05"),
+      period: 30n * 24n * 60n * 60n, // 30 days
+    };
+    const custodianVaultParameters = {
+      partialAuctionThreshold: custodianFee.amount * 15n,
+      partialAuctionDuration: custodianFee.period / 2n,
+      liquidationThreshold: custodianFee.amount * 2n,
+      newFractionsPerAuction: fractionsAmount * 5n,
+    };
+    let accounts: string[];
+    let balances: bigint[];
+
+    beforeEach(async function () {
+      // Mint fractions, but then manually delete the state to simulate v1.0.1 state
+      await fermionFNFTProxy
+        .connect(seller)
+        .mintFractions(
+          startTokenId,
+          1,
+          fractionsAmount,
+          auctionParameters,
+          custodianVaultParameters,
+          additionalDeposit,
+          ZeroAddress,
+        );
+
+      const fermionFnftAddress = await fermionFNFTProxy.getAddress();
+      const erc20FractionsAddress = await fermionFNFTProxy.getERC20CloneAddress(0);
+
+      // delete the FermionFNFT state
+      const fermionFractionSlotNumber = BigInt("0x4a7c305e00776741ac7013c3447ca536097b753ba0aa5e566dd79e90f6126200"); // keccak256(abi.encode(uint256(keccak256("fermion.fractions.storage")) - 1)) & ~bytes32(uint256(0xff))
+      const epochToCloneSlot = BigInt(keccak256("0x" + fermionFractionSlotNumber.toString(16)));
+      const currentEpochSlot = fermionFractionSlotNumber + 1n;
+
+      await setStorageAt(fermionFnftAddress, fermionFractionSlotNumber, ZeroHash); // set epochToClone length to 0
+      await setStorageAt(fermionFnftAddress, epochToCloneSlot, ZeroHash); // set first epoch to clone address to 0
+      await setStorageAt(fermionFnftAddress, currentEpochSlot, ZeroHash); // set currentEpoch to 0
+
+      // delete the ERC20 contract
+      await ethers.provider.send("hardhat_setCode", [await erc20FractionsAddress, "0x"]);
+
+      // set up the balances
+      const erc20StorageSlotNumber = BigInt("0x52c63247e1f47db19d5ce0460030c497f067ca4cebf71ba98eeadabe20bace00"); // keccak256(abi.encode(uint256(keccak256("openzeppelin.storage.ERC20")) - 1)) & ~bytes32(uint256(0xff))
+      accounts = wallets.slice(3, 10).map((wallet) => wallet.address);
+      accounts.push(await fermionFNFTProxy.getAddress());
+      balances = [...Array(Number(accounts.length)).keys()].map((n) => parseEther((n + 1).toString()));
+
+      for (let i = 0; i < accounts.length; i++) {
+        const balanceSlot = BigInt(keccak256(toBeHex(accounts[i], 32) + erc20StorageSlotNumber.toString(16)));
+        await setStorageAt(fermionFnftAddress, balanceSlot, toBeHex(balances[i].toString(), 32));
+      }
+    });
+
+    it("Migrate all accounts", async function () {
+      expect(await fermionFNFTProxy.getERC20CloneAddress(0)).to.equal(ZeroAddress);
+
+      const tx = await fermionFNFTProxy.migrateFractions(accounts.slice(0, -1)); // do not pass in the fnft contract address
```

Comment:
Question: in reality, how do we get the list of fraction "owners"?
I guess we can scan all Transfer events emitted from the FermionFractions contract and build a list of owners (all accounts with a >0 balance in the end)